src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -4368,7 +4368,10 @@ OGRLayerH QgsOgrProviderUtils::setSubsetString( OGRLayerH layer, GDALDatasetH ds
   }
   else
   {
-    OGR_L_SetAttributeFilter( layer, encoding->fromUnicode( subsetString ).constData() );
+    if ( OGR_L_SetAttributeFilter( layer, encoding->fromUnicode( subsetString ).constData() ) != OGRERR_NONE )
+    {
+      return nullptr;
```

Comment:
Any chance to also log an error message if this happens?

Reply:
`QgsOgrProviderUtils::setSubsetString` is just a utility method used by `QgsOgrProvider::_setSubsetString` that handles the error and the logging:
https://github.com/qgis/QGIS/blob/0b4cc5adaeed099f83a47a527acefa6e2ac8bd86/src/core/providers/ogr/qgsogrprovider.cpp#L2067-L2073
then
https://github.com/qgis/QGIS/blob/0b4cc5adaeed099f83a47a527acefa6e2ac8bd86/src/core/qgsvectordataprovider.cpp#L705-L710

So I don't think extra logging is needed. I could add some unit test though, I just need to figure out where!...

Reply:
That sounds good.

Some tests exist already in `tests/src/python/test_provider_ogr*.py`, you could build on top of that

Reply:
Yes, but shapefile and spatialite tests are in `test_provider_shapefile.py` and `test_provider_spatialite.py`, are they not using ogrdataprovider? I think I'll put them on `test_provider_ogr.py` that is more generic, though currently all tests regarding `setSubsetString` are in each format's test file.

Reply:
Changed my mind and added the test on `test_provider_shapefile.py` since most ogr tests are there.